staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/v2/conversion.go

```diff
@@ -81,6 +81,13 @@ func ToStructuralOpenAPIV2(in *structuralschema.Structural) *structuralschema.St
 				changed = true
 			}
 
+			if s.XPreserveUnknownFields && s.Type == "object" {
```

Comment:
`&& additionalProperties is not set`, right?

Reply:
The bug is still present if `additionalProperties` is set I think.

- You can reproduce it with `additionalProperties` sets to `true`:

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: foos.k8s.test
spec:
  group: k8s.test
  versions:
    - name: v1
      served: true
      storage: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            field:
              type: object
              x-kubernetes-preserve-unknown-fields: true
              additionalProperties: true
  scope: Namespaced
  names:
    plural: foos
    singular: foo
    kind: Foo
```

```yaml
apiVersion: k8s.test/v1
kind: Foo
metadata:
  name: foo
field:
  foo: null
```

- You can reproduce it with `additionalProperties` including `nullable: true`:

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: foos.k8s.test
spec:
  group: k8s.test
  versions:
    - name: v1
      served: true
      storage: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            field:
              type: object
              x-kubernetes-preserve-unknown-fields: true
              additionalProperties:
                type: object
                nullable: true
  scope: Namespaced
  names:
    plural: foos
    singular: foo
    kind: Foo
```

```yaml
apiVersion: k8s.test/v1
kind: Foo
metadata:
  name: foo
field:
  foo: null
```

Reply:
ack